source/common/http/filter_manager.cc

```diff
@@ -1196,9 +1223,29 @@ void FilterManager::encodeMetadata(ActiveStreamEncoderFilter* filter,
       return;
     }
 
+    state_.filter_call_state_ |= FilterCallState::EncodeMetadata;
+
     FilterMetadataStatus status = (*entry)->handle_->encodeMetadata(*metadata_map_ptr);
-    ENVOY_STREAM_LOG(trace, "encode metadata called: filter={} status={}", *this,
-                     (*entry)->filter_context_.config_name, static_cast<uint64_t>(status));
+
+    state_.filter_call_state_ &= ~FilterCallState::EncodeMetadata;
+
+    ENVOY_STREAM_LOG(trace, "encode metadata called: filter={} status={} metadata={}", *this,
+                     (*entry)->filter_context_.config_name, static_cast<uint64_t>(status),
+                     *metadata_map_ptr);
+
+    if (status == FilterMetadataStatus::StopIterationForLocalReply) {
+      // The way is shut. It was made by those who are dead, and the dead keep it.
```

Comment:
Can you illuminate this more?  Why the asymmetry?

Reply:
Seem like this should also not just fail silently?

Reply:
This still fails silently. Can you provide context why we don't want to support this on the encode* route?

Reply:
makes sense